### DISCUSSION TITLE
Marking text by several whole words for copying or annotating

### DISCUSSION THREAD

**Discussion by ShieldnSword** [sumatrapdfreader/sumatrapdf#4022] [Ideas]
Marking text by only 1 whole word is possible by double-clicking at version 3.5.2/64bit. But it's not possible to mark several whole words by double-clicking on the 1. word, hold the mouse button at the 2. click and move the cursor over the next words. The text marking on the 1. word isn't extended.
The later is possible e.g. in text processors, e-mail clients, table sheets and this feature would improve the speed and comfort at working.

**Comment by GitHubRulesOK** [comment]
The way text is stored inside a pdf is simply single letters and groups (possibly with or without spaces)  along a single line.

So the way a word selector functions is, it sees the letter you click then scans left and right at same height until there is an end to characters in a range of A-Z+0-9 (and a few other bytes) without going to the edge of the page.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/f834ea1e-9065-4286-92f7-4b8eb70c1623)
Thus punctuation including a spacebar or `?` is considered end of line. This is very simple 1st level but works for most single western words.

The other problem is PDF text is not the way you might expect from a word processor.
so If I select the punctuation It may not include the letters as they may have been written before or after !  

![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/16562b37-f21d-40fb-8f14-2da622f5c62a)
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/575171cc-933c-4f21-a16a-5e4ebd50fab4)


The logic to scan and highlight an area between 2 points is also easy by avoiding letters as anchors.
![image](https://github.com/sumatrapdfreader/sumatrapdf/assets/12754913/8432fb97-9db8-4f9e-94b8-3cfe1988f1ad)

So the highlight or selection is EXACTLY the same when scanned by letters or area the way PDF holds annotation is NOT locked to single letters, it simply looks that way when using word selection.
